openquake/commonlib/readinput.py

```diff
@@ -962,13 +955,18 @@ def get_sitecol_assetcol(oqparam, haz_sitecol=None, cost_types=()):
         discarded = []
         logging.info('Read %d sites and %d assets from the exposure',
                      len(sitecol), sum(len(a) for a in assets_by_site))
-
-    asset_refs = numpy.array(
-        [exposure.asset_refs[asset.ordinal]
-         for assets in assets_by_site for asset in assets])
     assetcol = asset.AssetCollection(
-        asset_refs, assets_by_site, exposure.tagcol, exposure.cost_calculator,
-        oqparam.time_event, exposure.occupancy_periods)
+        exposure, assets_by_site, oqparam.time_event)
+    if assetcol.occupancy_periods:
+        missing = set(cost_types) - set(exposure.cost_types['name']) - set(
+            ['occupants'])
+    else:
+        missing = set(cost_types) - set(exposure.cost_types['name'])
+    if missing and not oqparam.calculation_mode.endswith('damage'):
+        expo = oqparam.inputs.get('exposure', '')
+        raise InvalidFile(
+            'Expected cost types %s but the exposure %r contains %s' % (
```

Comment:
Wouldn't it be better to point out the missing data in the exposure directly? Otherwise we could end up with a message like the following: 
```sh
Expected cost types ['occupants', 'structural'] but the exposure ['exposure_model.xml']
contains ['business_interruption' 'contents' 'nonstructural' 'structural']
```

Reply:
Done